cumulusci/tasks/push/tests/test_push_tasks.py

```diff
@@ -0,0 +1,481 @@
+    with open(ORG_FILE, "w") as file:
+        file.write(f"{ORG_FILE_TEXT}")
+    task = create_task(
+        SchedulePushOrgList,
+        options={
+            "orgs": ORG_FILE,
+            "version": VERSION,
+            "namespace": NAMESPACE,
+            "start_time": datetime.datetime.now(),
+            "batch_size": 10,
+        },
+    )
+    assert task._get_orgs() == ["00DS0000003TJJ6MAO", "00DS0000003TJJ6MAL"]
+    os.remove(ORG_FILE)
+
+
+def test_schedule_push_org_list_init_options():
+    with open(ORG_FILE, "w") as file:
+        file.write(f"{ORG_FILE_TEXT}")
+    task = create_task(
+        SchedulePushOrgList,
+        options={
+            "orgs": ORG_FILE,
+            "version": VERSION,
+            "start_time": datetime.datetime.now(),
+        },
+    )
+    task._init_task()
+    task._init_options(
+        {"orgs": ORG_FILE, "version": VERSION, "start_time": datetime.datetime.now()}
+    )
+    assert task.options["namespace"] is None
+    assert task.options["batch_size"] == 200
+    assert task.options["orgs"] == ORG_FILE
+    assert task.options["version"] == VERSION
+    os.remove(ORG_FILE)
+
+
+def test_schedule_push_org_list_bad_start_time():
+    with open(ORG_FILE, "w") as file:
+        file.write(f"{ORG_FILE_TEXT}")
+    task = create_task(
+        SchedulePushOrgList,
+        options={
+            "orgs": ORG_FILE,
+            "version": VERSION,
+            "start_time": "2020-06-10T10:15",
+            "namespace": NAMESPACE,
+        },
+    )
+    task.push = mock.MagicMock()
+    with pytest.raises(CumulusCIException):
+        task._run_task()
+    os.remove(ORG_FILE)
+
+
+def test_load_orgs_file():
+    # creating sample org file for testing
+    # testing with empty file
+    with open(ORG_FILE, "w") as file:
+        file.write(f"{ORG_FILE_TEXT}")
+        task = create_task(BaseSalesforcePushTask, options={})
+        assert task._load_orgs_file(ORG_FILE) == []
+
+    # testing with multiple orgs
+    with open(ORG_FILE, "r") as file:
+        assert task._load_orgs_file(ORG_FILE) == [
+            "00DS0000003TJJ6MAO",
+            "00DS0000003TJJ6MAL",
+        ]
+    os.remove(ORG_FILE)
+
+
+def test_get_subs_raises_err():
+    task = create_task(GetSubscriberList, options={"filename": "in.csv"})
+    with pytest.raises(NotImplementedError):
+        task()
+
+
+def test_filter_subs_raises_err():
+    task = create_task(
+        FilterSubscriberList, options={"file_in": "in.csv", "file_out": "out.txt"}
+    )
+    with pytest.raises(NotImplementedError):
+        task()
+
+
+def test_base_push_task_raises_err():
+    task = create_task(BaseSalesforcePushTask, options={})
+    with pytest.raises(NotImplementedError):
+        task()
+
+
+def test_report_push_status_error():
+    task = create_task(BaseSalesforcePushTask, options={})
+    task.sf = mock.MagicMock()
+    task.push_report = mock.MagicMock()
+    task.sf.query_all.return_value = {"totalSize": 1, "records": []}
```

Comment:
This is a nonsensical query result -- if records is an empty list, then totalSize would be 0